contracts/MemeBase.sol

```diff
@@ -157,30 +158,41 @@ contract MemeBase is MemeFactory {
     function _redemptionSetup(address[] memory accounts, uint256[] memory amounts) private {
         require(accounts.length == amounts.length);
 
-        redemptionAddress = address(new Meme("Agent Token", "AGENT", DECIMALS, MIN_TOTAL_SUPPLY));
+        redemptionAddress = address(new Meme("Agent Token II", "AGNT II", DECIMALS, 1_000_000_000 ether));
 
         // Record all the accounts and amounts
         uint256 totalAmount;
         for (uint256 i = 0; i < accounts.length; ++i) {
-            // Adjust amount for already collected burned tokens
-            uint256 adjustedAmount = (amounts[i] * 9) / 10;
-            totalAmount += adjustedAmount;
-            memeHearters[redemptionAddress][accounts[i]] = adjustedAmount;
+            totalAmount += amounts[i];
+            memeHearters[redemptionAddress][accounts[i]] = amounts[i];
+            // to match original hearter events
+            emit Hearted(accounts[i], redemptionAddress, amounts[i]);
         }
+        require(totalAmount == CONTRIBUTION_AGNT, "Total amount must match original contribution amount");
+        // Adjust amount for already collected burned tokens
+        uint256 adjustedAmount = (totalAmount * 9) / 10;
+        require(adjustedAmount == REDEMPTION_AMOUNT, "Total amount adjusted for burn allocation must match redemption amount");
 
         // summonTime is set to zero such that no one is able to heart this token
-        memeSummons[redemptionAddress] = MemeSummon(REDEMPTION_AMOUNT, 0, 0, 0);
+        memeSummons[redemptionAddress] = MemeSummon(adjustedAmount, 0, 0, 0);
```

Comment:
This now becomes incorrect. You are putting a total as `(totalAmount * 9) / 10`, but recording individual heart as the original `amounts[i]` for each. Let's be then consistent and put `CONTRIBUTION_AGNT` here.